tools/utils/fiber-helpers.ts

```diff
@@ -126,18 +125,22 @@ exports.bindEnvironment = function (func) {
     if (Fiber.current) {
       return runWithEnvironment();
     }
+
     Fiber(runWithEnvironment).run();
   };
 };
 
 // Returns a Promise that supports .resolve(result) and .reject(error).
-exports.makeFulfillablePromise = function () {
-  var resolve, reject;
-  var promise = new Promise(function (res, rej) {
+export function makeFulfillablePromise() {
+  let resolve: (value?: any) => void = () => { };
+  let reject: (reason?: any) => void = () => { };
```

Comment:
I didn't check where these type of promises are used but I'm not sure if changing and undefined variable for a noop maintains the assumptions about the `resolve` and `reject` properties. For example, what if a piece of code would rely on checking whether the `resolve` is defined on a particular promise.

Reply:
Thanks. I fixed it. But maybe you know more graceful solution?

Reply:
Unfortunately I've also had to make transformations that are not elegant just to make sure that I don't break anything. For example:
```
function coalesce(delay) {
   delay = delay || 100
  // ... use delay
}
```
was transformed to:
```
function coalesce(delay: number) {
   const actualDelay = delay || 100
  // ... use actualDelay
}
```
instead of:
```
function coalesce(delay: number = 100) {
  // ... use delay
}
```
since the behavior would be changed if coalesce would get a `0` or a `null` instead of undefined. These are small things, and most of the time there's no difference, but (since I don't know the codebase very well) I wanted to stay on the safe side.